### ISSUE TITLE
AMP support for CodeFund ads

### ISSUE THREAD

**Issue by coderberry** [gitcoinco/code_fund_ads#1048]
From a publisher:

> The biggest challenge I faced was the ad code is not compatible with amp pages. So I could not display ads in my amp pages. A good part of my traffic is from amp pages. The only option for me was to go back to Google Adsense. Is there any way to get Codefund ad code that is compatible with AMP?

**Comment by andrewmcodes** [comment]
Need to investigate this more.

**Comment by andrewmcodes** [comment]
@coderberry This appears to work fine unless I'm missing something.

https://github.com/andrewmcodes/codefund-amp-demo/commit/1bba39b2c4b7f79949f305c42b52a6a47b64a55f

![The Photo Blog 2020-02-11 23-30-25](https://user-images.githubusercontent.com/18423853/74303044-82656300-4d26-11ea-8851-6df1e782e7f3.jpg)

**Comment by andrewmcodes** [comment]
@jdorfman tagging you as well in case you want to be in the loop on this

**Comment by jdorfman** [comment]
@andrewmcodes is that the only template you tried?

**Comment by andrewmcodes** [comment]
@jdorfman yes but the template should be irrelevant. I could just have an empty page with just the ad and this still should work. All you need is:

```html
<script async custom-element="amp-script" src="https://app.codefund.io/properties/###/funder.js"></script>
<amp-script id="codefund"></amp-script>
```

**Comment by jdorfman** [comment]
@andrewmcodes @coderberry I email the publisher who was wondering if we supported AMP.

I'll keep you posted.

**Comment by mishelshaji** [comment]
The code still has validation errors when I tested the page using google amp test.
https://search.google.com/test/amp
![Capture](https://user-images.githubusercontent.com/33393350/74510559-050a4000-4f2a-11ea-8b59-2a5b5ac1be88.PNG)

**Comment by jdorfman** [comment]
@mishelshaji thanks for sharing. @andrewmcodes any ideas?

**Comment by andrewmcodes** [comment]
Thanks for sharing that @mishelshaji, I didn’t know they needed to pass that test. I went back and did some more reading and this is actually a lot bigger than I thought. 

AMP doesn’t allow you to use custom scripts, which means we have no way to support this currently. Also from what I’ve read, it seems you have to create your creatives in a very special way specific to AMP, and it also appears that we wouldn’t be able to record impressions without tracking users. 

I’ll do some more due diligence here @jdorfman but unless I’m completely missing something that @mishelshaji can shed some light on, this isn’t something we can support unfortunately. Like I said- I’ll do some more research but wanted to let you know that we currently don’t have a way to support this.
